scripts/Test-Signatures.ps1

```diff
@@ -0,0 +1,297 @@
+[CmdletBinding()]
+param (
+    [String]
+    $Path = "$PSScriptRoot\packages",
+
+    [Switch]
+    $CIBuild
+)
+
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+
+# Allowed certificate subjects
+$script:ValidSubjects = @( 
+    "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" 
+    "CN=Microsoft Corporation, OU=MOPR, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" 
+)
+
+# Allowed certificate thumbprints
+$script:ValidThumbprints = @( 
+    "B9EAA034C821C159B05D3521BCF7FEB796EBD6FF" 
+    "98ED99A67886D020C564923B7DF25E9AC019DF26" 
+    "5EAD300DC7E4D637948ECB0ED829A072BD152E17" 
+    "67B1757863E3EFF760EA9EBB02849AF07D3A8080" 
+    "9DC17888B5CFAD98B3CB35C1994E96227F061675" 
+    "62009AAABDAE749FD47D19150958329BF6FF4B34" 
+    "3BDA323E552DB1FDE5F4FBEE75D6D5B2B187EEDC" 
+    "899FA016DEE8E665FF2A315A1151C43FB96C430B"  # Microsoft 3rd Party Application Component
+)
+
+# Allow-list for unsigned binaries
+# No binaries allow to ship without a valid signature as per policy
+$script:UnsignedAllowList = @{
+    # "PackageId" = relative paths
+    # "Microsoft.TestPlatform" = @( `
+    #     ".\tools\net451\Common7\IDE\Extensions\TestPlatform\Newtonsoft.Json.dll" `
+    # )
+}
+
+$script:Depth = 0
+
+function Get-RelativePath($Path, $RelativeTo) {
+    try {
+        Push-Location
+        Set-Location $RelativeTo
+        $Path = Resolve-Path $Path -Relative
+    }
+    finally {
+        Pop-Location
+    }
+
+    $Path
+}
+
+function Push-Depth {
+    $script:Depth = $script:Depth + 1
+}
+
+function Pop-Depth {
+    $script:Depth = $script:Depth - 1
+
+    if($script:Depth -lt 0) {
+        $script:Depth = 0
+    }
+}
+
+function Clear-Line($Message = "") {
+    if($CIBuild) {
+        return
+    }
+    
+    $x = [System.Console]::get_CursorLeft()
+    Write-Host "`r$(' ' * $x)$Message`r" -NoNewLine
+}
+
+function Update-Operation($Operation, $Message, [switch]$NewLine, [switch]$Suppress) {

```

Comment:
I commented on this before. Why is this not using the built in Write-Progress functionality?